### DISCUSSION TITLE
Dynamic Links - how to clear once retrieved?

### DISCUSSION THREAD

**Discussion by Zelfapp** [firebase/flutterfire#6274] [Q&A]
I have dynamic links working great. I can get query parameters and route to paths. However, is there a way to clear a dynamic link once it has been retrieved?

E.g. when I retrieve the dynamic link (regardless of whether app is closed or in the background) I load the appropriate deep link path. Then, if I try to go to another screen that has an `initState` that is also looking for dynamic links, the same dynamic link that was previously found and processed is found again and will create a loop where I cannot get away from the dynamic link path.

Would be nice to have a `FirebaseDynamicLinks.instance.clear()` type of function to avoid getting in these loops, unless I'm missing something already in the docs.

If there is a not a "clear()" function of some kind, how do I keep track of what dynamic links have already been processed?

**Comment by Vaytrex** [comment]
For anyone landing here after the Aug 2025 shutdown: there's no drop-in. You need a URL that branches on User-Agent (App Store / intent:// with browser_fallback_url / website) and skips redirecting crawlers so previews survive. Full write-up with the intent format: https://linkbah.app/firebase-dynamic-links-alternative
